cupy/cuda/stream.pyx

```diff
@@ -339,7 +339,7 @@ class Stream(BaseStream):
         if is_shutting_down():
             return
         tls = _ThreadLocal.get()
-        if self.ptr:
+        if self.ptr and self.ptr != runtime.streamPerThread:
```

Comment:
Just a nitpick: Can we also include `streamLegacy` here to be future-proof? See
https://github.com/cupy/cupy/blob/8cf5983112a4e90fe5e83b8daec5e55da472f400/cupy/cuda/stream.pyx#L322-L325
```suggestion
        if self.ptr not in (0, 1, 2):
```